src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -231,12 +236,33 @@ headers_iterator(void *data, enum MHD_ValueKind kind, const char *key, const cha
 }
 
 static int
+get_static_file(const char *dir, const char *url)
+{
+    int ret;
+    char path[PATH_MAX];
+
+    /* url given by microhttpd starts from /. For instance,
+     * https://www.solettaproject.com => url == /
+     * https://www.solettaproject.com/thankyou => url == /thankyou
+     */
+    ret = snprintf(path, sizeof(path), "%s/%s", dir,
+        *(url + 1) ? (url + 1) : "index.html");
+    SOL_INT_CHECK(ret, < 0, NULL);
+
+    /*  According with microhttpd fd will be closed when response is
+     *  destroyed and fd should be in 'blocking' mode
+     */
+    return open(path, O_RDONLY | O_CLOEXEC);
```

Comment:
Call `fstat(fd, &st)` and bail out if `st.st_mode & (S_IRUSR | S_IRGRP | S_IROTH)` fails.